runtime/src/main/java/org/corfudb/runtime/view/AbstractView.java

```diff
@@ -58,24 +60,39 @@ public Layout getCurrentLayout() {
         }
     }
 
+    public <T, A extends RuntimeException, B extends RuntimeException, C extends RuntimeException,
+            D extends RuntimeException> T layoutHelper(LayoutFunction<Layout, T, A, B, C, D>
+                                                               function)
+            throws A, B, C, D {
+        return layoutHelper(function, false);
+    }
+
+
     /**
      * Helper function for view to retrieve layouts.
      * This function will retry the given function indefinitely,
      * invalidating the view if there was a exception contacting the endpoint.
      *
+     * There is a flag to set if we want the caller to handle Runtime Exceptions. For some
+     * special cases (like writes), we need to do a bit more work upon a Runtime Exception than just retry.
+     *
      * @param function The function to execute.
      * @param <T>      The return type of the function.
      * @param <A>      Any exception the function may throw.
      * @param <B>      Any exception the function may throw.
      * @param <C>      Any exception the function may throw.
      * @param <D>      Any exception the function may throw.
+     * @param rethrowAllExceptions if all exceptions are rethrown to caller.
      * @return The return value of the function.
      */
     public <T, A extends RuntimeException, B extends RuntimeException, C extends RuntimeException,
             D extends RuntimeException> T layoutHelper(LayoutFunction<Layout, T, A, B, C, D>
-                                                                          function)
+                                                                          function,
+                                                       boolean rethrowAllExceptions)
             throws A, B, C, D {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'A' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'B' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'C' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'D' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)